og.module

```diff
@@ -306,3 +313,64 @@ function og_invalidate_group_content_cache_tags(EntityInterface $entity) {
     Cache::invalidateTags(Cache::buildTags('og-group-content', $tags));
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_insert() for OgRole entities.
+ */
+function og_og_role_insert(OgRoleInterface $role) {
+  // Create actions to add or remove the role, except for the required default
+  // roles 'member' and 'non-member'. These cannot be added or removed.
+  if ($role->getRoleType() === OgRoleInterface::ROLE_TYPE_REQUIRED) {
+    return;
+  }
+
+  $add_id = 'og_membership_add_single_role_action.' . $role->getName();
+  if (!Action::load($add_id)) {
+    $action = Action::create([
+      'id' => $add_id,
+      'type' => 'og_membership',
+      'label' => t('Add the @label role to the selected members', ['@label' => $role->label()]),
```

Comment:
Very good find, since roles for different groups can have different labels this makes sense. We cannot use the labels unless we know on which group(s) they apply.

Unfortunately we do not have the right context available to know on which group(s) the views bulk operations take effect. In the default view that we supply with OG the group type and group ID are part of the contextual arguments, but we cannot rely on these being always available. Views are very complex beasts, it is possible to limit the group using regular filters, or even `hook_views_query_alter()`.

I agree that using the role name (machine name) is not the prettiest solution but it will at least be 100% clear and correct.